core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -0,0 +1,5534 @@
+    this.stringEncoding = encoding;
+    determineSingleByteEncoding();
+    compareStorageAndActualFormat();
+  }
+
+  /**
+   * @return the decimalSymbol
+   */
+  @Override
+  public String getDecimalSymbol() {
+    return decimalSymbol;
+  }
+
+  /**
+   * @param decimalSymbol
+   *          the decimalSymbol to set
+   */
+  @Override
+  public void setDecimalSymbol( String decimalSymbol ) {
+    this.decimalSymbol = decimalSymbol;
+    decimalFormatChanged = true;
+    compareStorageAndActualFormat();
+  }
+
+  /**
+   * @return the groupingSymbol
+   */
+  @Override
+  public String getGroupingSymbol() {
+    return groupingSymbol;
+  }
+
+  /**
+   * @param groupingSymbol
+   *          the groupingSymbol to set
+   */
+  @Override
+  public void setGroupingSymbol( String groupingSymbol ) {
+    this.groupingSymbol = groupingSymbol;
+    decimalFormatChanged = true;
+    compareStorageAndActualFormat();
+  }
+
+  /**
+   * @return the currencySymbol
+   */
+  @Override
+  public String getCurrencySymbol() {
+    return currencySymbol;
+  }
+
+  /**
+   * @param currencySymbol
+   *          the currencySymbol to set
+   */
+  @Override
+  public void setCurrencySymbol( String currencySymbol ) {
+    this.currencySymbol = currencySymbol;
+    decimalFormatChanged = true;
+  }
+
+  /**
+   * @return the caseInsensitive
+   */
+  @Override
+  public boolean isCaseInsensitive() {
+    return caseInsensitive;
+  }
+
+  /**
+   * @param caseInsensitive
+   *          the caseInsensitive to set
+   */
+  @Override
+  public void setCaseInsensitive( boolean caseInsensitive ) {
+    this.caseInsensitive = caseInsensitive;
+  }
+
+  /**
+   * @return the collatorDisabled
+   */
+  @Override
+  public boolean isCollatorDisabled() {
+    return collatorDisabled;
+  }
+
+  /**
+   * @param collatorDisabled
+   *          the collatorDisabled to set
+   */
+  @Override
+  public void setCollatorDisabled( boolean collatorDisabled ) {
+    this.collatorDisabled = collatorDisabled;
+  }
+
+  @Override
+  public Locale getCollatorLocale() {
+    return this.collatorLocale;
+  }
+
+  /**
+   * @ sets the collator Locale
+   */
+  @Override
+  public void setCollatorLocale( Locale locale ) {
+    // Update the collator only if required
+    if ( collatorLocale == null || !collatorLocale.equals( locale ) ) {
+      this.collatorLocale = locale;
+      this.collator = Collator.getInstance( locale );
+    }
+  }
+
+  /**
+   * @get the collatorStrength
+   */
+  @Override
+  public int getCollatorStrength() {
+    return collatorStrength;
+  }
+
+  /**
+   * @param collatorStrength
+   *          the collatorStrength to set
+   */
+  @Override
+  public void setCollatorStrength( int collatorStrength ) throws IllegalArgumentException {
+    try {
+      if ( collator != null ) {
+        this.collator.setStrength( collatorStrength );
+        this.collatorStrength = collatorStrength;
+      }
+    } catch ( IllegalArgumentException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)